src/main/scala/viper/gobra/ast/internal/Program.scala

```diff
@@ -376,7 +376,7 @@ sealed trait Accessible extends Node {
 
 object Accessible {
   case class Predicate(op: PredicateAccess) extends Accessible with TriggerExpr
-  case class ExprAccess(op: Expr) extends Accessible
+  case class ExprAccess(op: Expr, underlyingType: Type) extends Accessible
```

Comment:
I do not think you need to add the underlying type since we can get this with the underlyingType function of the Context (which uses the LookupTable internally).